pyServer/AzureDiskInspectService.py

```diff
@@ -213,47 +233,133 @@ def ParseUrlArguments(self, urlPath, sasKey):
         return operationId, mode, modeMajorSkipTo, modeMinorSkipTo, storageAcctName, container_blob_name, storageUrl
 
     """
-    Upload a local file as a HTTP binary response.
+    Validate and parse the given destination Blob Sas Url.
+
+    A Blob Sas Url has these components:
+
+        https://<a. Account_Name>.<b. Blob_endpoint_suffix>/<c. Container_Name>/<d. Blob_Name>?<e. Sas_Token>
+
+    For example:
+
+        https://myaccount.blob.core.windows.net/mycontainer/outputfile.zip?some_sas_token
+
+    These can be parsed using the Python urllib.parse.urlparse tool by looking up:
+
+        urlparse(blobSasUrl).netloc = <a. Account_Name>.<b. Blob_endpoint_suffix>
+        urlparse(blobSasUrl).path = <c. Container_Name>/<d. Blob_Name>
+        urlparse(blobSasUrl).query: = <e. Sas_Token>
+
+    The following validating and parsing logic is based on the pattern described.
+    """
+    def ValidateAndParseBlobSasUrl(self, blobSasUrl):
+        self.telemetryLogger.info("Validating input destination Blob Sas Url.")
+
+        urlParts = urllib.parse.urlparse(blobSasUrl)
+
+        if not urlParts.path or len(urlParts.path) == 0:
+            raise InvalidBlobSasUrlException('Input Blob SAS Url for upload is missing Container and Blob info.')
+        if not urlParts.query or len(urlParts.query) == 0:
+            raise InvalidBlobSasUrlException('Input Blob SAS url for upload is missing SAS token.')
+
+        if urlParts.path[0] == '/':
+            blob_storage_path = urlParts.path[1:]
+        else:
+            blob_storage_path = urlParts.path
+
+        if 'blob' not in urlParts.netloc:
+            raise InvalidBlobSasUrlException('Input Blob SAS url for upload is not pointing to a valid Azure Blob endpoint.')
+
+        self.destination_storage_account = urlParts.netloc.split('.')[0]
+
+        if not self.destination_storage_account:
+            raise InvalidBlobSasUrlException('Input Blob SAS url for upload does not contain storage account.')
+
+        urlSplit = blob_storage_path.split('/')
+
+        if not len(urlSplit) == 2 or not urlSplit[0] or not urlSplit[1]:
+            raise InvalidBlobSasUrlException('Input Blob SAS Url is not in the right format.')
+        
+        self.destinaion_container_name = urlSplit[0]
+        self.destinaion_blob_name = urlSplit[1]
+        self.destination_sas_token = urlParts.query
+
+        self.telemetryLogger.info("Destination Blob Storage Account: " + self.destination_storage_account)
+        self.telemetryLogger.info("Container Name: " + self.destinaion_container_name)
+        self.telemetryLogger.info("Blob Name: " + self.destinaion_blob_name)
+
     """
-    def uploadFile(self, http_headers, outputFileName, isPartial, osType):
+    Upload the given file to destination Blob storage using the given Sas Url.
+    """
+    def uploadFileWithBlobSasUrl(self, file_name_full_path):
```

Comment:
Might be good do have some retry logic here.  Given we just did a ton of work to gather the payload, it would not be good if it was wasted on some intermittent issue with blob storage.

Reply:
Added retry logic for this.